tests/Jobs/PtpJobTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Biigle\Tests\Modules\Ptp\Jobs;
+
+use Biigle\ImageAnnotation;
+use Biigle\ImageAnnotationLabel;
+use Biigle\Modules\Ptp\Exceptions\PythonException;
+use Biigle\Modules\Ptp\Jobs\PtpJob;
+use Biigle\Shape;
+use Biigle\Image;
+use Biigle\Label;
+use Biigle\User;
+use Biigle\Volume;
+use Ramsey\Uuid\Uuid;
+use Storage;
+use TestCase;
+
+
+class PtpJobTest extends TestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->volume = Volume::factory()->create();
+        $this->image = Image::factory()->create(['volume_id' => $this->volume->id]);
+        $attrs = $this->volume->attrs;
+        $this->uuid = Uuid::uuid4();
+        $attrs['ptp_job_id'] = $this->uuid;
+        $this->volume->attrs = $attrs;
+        $this->volume->save();
+        $this->imageAnnotation = ImageAnnotation::factory()->create([
+            'image_id' => $this->image->id,
+        ]);
+        $this->label = Label::factory()->create();
+        $this->user = User::factory()->create();
+        $this->user2 = User::factory()->create();
+        $this->imageAnnotationLabel = ImageAnnotationLabel::factory()->create([
+            'annotation_id' => $this->imageAnnotation->id,
+            'label_id' => $this->label->id,
+            'user_id' => $this->user->id,
+        ]);
+        $this->inputFile = 'test';
+        $inputFileContent = [
+            $this->image->id => [[
+                'annotation_id' => $this->imageAnnotation->id,
+                'points' => [1,2],
+                'shape' => Shape::pointId(),
+                'image' => $this->image->id,
+                'label' => $this->label->id,
+            ]]];
+        $this->outputFile = 'testOutput.json';
+        $outputFileContent = [[
+            'points' => [1,2,3,4,5,6],
+            'annotation_id' => $this->imageAnnotation->id,
+            'label_id' => $this->label->id,
+        ]];
+
+        config(['ptp.ptp_storage_disk' => 'test']);
+        config(['ptp.device' => 'cpu']);
+        $storage = Storage::disk('test');
+        $storage->put(config('ptp.temp_dir').'/'.$this->inputFile.'.json', json_encode($inputFileContent));
+        file_put_contents(config('ptp.temp_dir').'/'.$this->outputFile, json_encode($outputFileContent));
+    }
```

Comment:
Some of this is not required in all of the tests (as far as I can see). Particularly creating files on disk should be avoided wherever possible as it makes the tests slow. But *when* you create files, you also have to take care of cleaning them up after the test or when the test fails. I usually do it like this:

```php
// create file
try {
   // test stuff
} finally {
   // delete file
}
```